### ISSUE TITLE
Wakatime-cli @Latest AND @Release, unable to send heartbeat.

### ISSUE THREAD

**Issue by anoduck** [wakatime/wakatime-cli#837]
**Expected behavior (what should have happened)**:

Routine heartbeat update should have contacted the server without issues or error messages, and logged activity for the client.

**Actual behavior (what went wrong)**:

The issue was noticed while trying to write some code in Emacs. Wakatime-cli was unable to submit heartbeat to wakatime servers. Current client was updated from most current release, and then afterward built from current dev branch of Wakatime repo. All three resulted in the same explosion of error messages. Syntax of command execution was checked as well, and appears good. 

**Environment**:

- OS:  OpenBSD CURRENT 7.2
- Platform:  AMD64

**Logs**:

Oddly, `~/.wakatime.log` was absent of these errors, but they were dumped into the console for analysis.

```bash
SIGSYS: bad system call
PC=0x235f2e597 m=7 sigcode=0

goroutine 0 [idle]:
syscall.rawSyscall10X(0x4b1fc0, 0xc5, 0x0, 0x20000, 0x1, 0x1, 0x5, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/runtime/sys_openbsd3.go:114 +0x4d fp=0xc0001b20c0 sp=0xc0001b20a0 pc=0x462ecd
syscall.rawSyscall10X(0xd70edbc9210d3393?, 0xd0a853ee50b342d4?, 0x3bca7808b07fc9ba?, 0x6b685517610d0966?, 0x2a03f2acd3017326?, 0x17345fa8037da625?, 0x8bcc5a5b4a1dfbd5?, 0xc8a83467ef5f13ed?, 0x691bd42df67f76cc?, 0x0, ...)
	<autogenerated>:1 +0x59 fp=0xc0001b2140 sp=0xc0001b20c0 pc=0x4689d9
syscall.syscall9Internal(0x4?, 0x0?, 0x8?, 0x0?, 0xc0001b2218?, 0x440911?, 0x2?, 0x0?, 0xc0000061a0?, 0x0)
	/usr/local/go/src/syscall/syscall_openbsd_libc.go:38 +0x49 fp=0xc0001b21a8 sp=0xc0001b2140 pc=0x4ac9a9
syscall.syscall9Internal(0xc5, 0x0, 0x20000, 0x1, 0x1, 0x5, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:1 +0x68 fp=0xc0001b2208 sp=0xc0001b21a8 pc=0x4b23c8
golang.org/x/sys/unix.mmap(0x4b1c20?, 0x5?, 0xc00011fcb8?, 0x0?, 0x0?, 0x0?)
golang.org/x/sys/unix.(*mmapper).Mmap(0x13164c0, 0xc0001b2390?, 0x85b1f9?, 0x20000, 0xc0001b2390?, 0x4db1af?)
golang.org/x/sys/unix.Mmap(...)
go.etcd.io/bbolt.mmap(0xc000127440, 0x20000)
go.etcd.io/bbolt.(*DB).mmap(0xc000127440, 0x0)
go.etcd.io/bbolt.Open({0xc000660b80, 0x1d}, 0x5?, 0xc0001b3600)
github.com/wakatime/wakatime-cli/pkg/offline.popHeartbeats({0xc000660b80?, 0x40eadf?}, 0x203000?)
	/wakatime-cli/pkg/offline/offline.go:237 +0xb7 fp=0xc0001b3670 sp=0xc0001b3528 pc=0x86b6f7
github.com/wakatime/wakatime-cli/pkg/offline.Sync.func1(0xc00007e840)
	/wakatime-cli/pkg/offline/offline.go:144 +0xca fp=0xc0001b3748 sp=0xc0001b3670 pc=0x86aa6a
github.com/wakatime/wakatime-cli/pkg/offline.WithSync.func1.1({0xc00007e840?, 0xc48320?, 0xc0001b3701?})
	/wakatime-cli/pkg/offline/offline.go:108 +0xd3 fp=0xc0001b37b8 sp=0xc0001b3748 pc=0x86a833
github.com/wakatime/wakatime-cli/pkg/heartbeat.NewHandle.func1({0x0, 0x0, 0x0})
	/wakatime-cli/pkg/heartbeat/heartbeat.go:168 +0xd1 fp=0xc0001b3808 sp=0xc0001b37b8 pc=0x716ad1
github.com/wakatime/wakatime-cli/cmd/offlinesync.SyncOfflineActivity(0x2d1294848?, {0xc000660b80, 0x1d})
	/wakatime-cli/cmd/offlinesync/offlinesync.go:75 +0x3d3 fp=0xc0001b3ad0 sp=0xc0001b3808 pc=0xb9e693
github.com/wakatime/wakatime-cli/cmd/offlinesync.Run(0xe67a20?)
	/wakatime-cli/cmd/offlinesync/offlinesync.go:28 +0x85 fp=0xc0001b3b48 sp=0xc0001b3ad0 pc=0xb9e165
github.com/wakatime/wakatime-cli/cmd.runCmd(0xc00015b380, 0x0, 0x0, 0xd75d60)
	/wakatime-cli/cmd/run.go:268 +0x114 fp=0xc0001b3c10 sp=0xc0001b3b48 pc=0xba1b54
github.com/wakatime/wakatime-cli/cmd.RunCmdWithOfflineSync(0xc0003c58d0?, 0x9e?, 0xb2?, 0x3c?)
	/wakatime-cli/cmd/run.go:244 +0x4b fp=0xc0001b3c48 sp=0xc0001b3c10 pc=0xba19eb
github.com/wakatime/wakatime-cli/cmd.Run(0x0?, 0x0?)
	/wakatime-cli/cmd/run.go:109 +0x5d7 fp=0xc0001b3d70 sp=0xc0001b3c48 pc=0xba0f17
github.com/wakatime/wakatime-cli/cmd.NewRootCMD.func1(0xc000353900?, {0xca59a0?, 0x8?, 0x8?})
	/wakatime-cli/cmd/root.go:29 +0x1d fp=0xc0001b3d90 sp=0xc0001b3d70 pc=0xb9f71d
github.com/spf13/cobra.(*Command).execute(0xc000353900, {0xc0000240a0, 0x8, 0x8})
github.com/spf13/cobra.(*Command).ExecuteC(0xc000353900)
github.com/spf13/cobra.(*Command).Execute(...)
github.com/wakatime/wakatime-cli/cmd.Execute()
	/wakatime-cli/cmd/root.go:260 +0x1e fp=0xc0001b3f70 sp=0xc0001b3f20 pc=0xba08de
main.main()
	/wakatime-cli/main.go:6 +0x17 fp=0xc0001b3f80 sp=0xc0001b3f70 pc=0xba2a97
runtime.main()
	/usr/local/go/src/runtime/proc.go:250 +0x1f8 fp=0xc0001b3fe0 sp=0xc0001b3f80 pc=0x438a38
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0001b3fe8 sp=0xc0001b3fe0 pc=0x466221

goroutine 1 [running]:
runtime.systemstack_switch()
	/usr/local/go/src/runtime/asm_amd64.s:459 fp=0xc0001b2070 sp=0xc0001b2068 pc=0x463fe0
runtime.libcCall(0x718c67f7a6c8fc6?, 0x289e0406de478e3f?)
	/usr/local/go/src/runtime/sys_libc.go:49 +0x66 fp=0xc0001b20a0 sp=0xc0001b2070 pc=0x454286
syscall.rawSyscall10X(0x4b1fc0, 0xc5, 0x0, 0x20000, 0x1, 0x1, 0x5, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/runtime/sys_openbsd3.go:114 +0x4d fp=0xc0001b20c0 sp=0xc0001b20a0 pc=0x462ecd
syscall.rawSyscall10X(0xd70edbc9210d3393?, 0xd0a853ee50b342d4?, 0x3bca7808b07fc9ba?, 0x6b685517610d0966?, 0x2a03f2acd3017326?, 0x17345fa8037da625?, 0x8bcc5a5b4a1dfbd5?, 0xc8a83467ef5f13ed?, 0x691bd42df67f76cc?, 0x0, ...)
	<autogenerated>:1 +0x59 fp=0xc0001b2140 sp=0xc0001b20c0 pc=0x4689d9
syscall.syscall9Internal(0x4?, 0x0?, 0x8?, 0x0?, 0xc0001b2218?, 0x440911?, 0x2?, 0x0?, 0xc0000061a0?, 0x0)
	/usr/local/go/src/syscall/syscall_openbsd_libc.go:38 +0x49 fp=0xc0001b21a8 sp=0xc0001b2140 pc=0x4ac9a9
syscall.syscall9Internal(0xc5, 0x0, 0x20000, 0x1, 0x1, 0x5, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:1 +0x68 fp=0xc0001b2208 sp=0xc0001b21a8 pc=0x4b23c8
golang.org/x/sys/unix.mmap(0x4b1c20?, 0x5?, 0xc00011fcb8?, 0x0?, 0x0?, 0x0?)
golang.org/x/sys/unix.(*mmapper).Mmap(0x13164c0, 0xc0001b2390?, 0x85b1f9?, 0x20000, 0xc0001b2390?, 0x4db1af?)
golang.org/x/sys/unix.Mmap(...)
go.etcd.io/bbolt.mmap(0xc000127440, 0x20000)
go.etcd.io/bbolt.(*DB).mmap(0xc000127440, 0x0)
go.etcd.io/bbolt.Open({0xc000660b80, 0x1d}, 0x5?, 0xc0001b3600)
github.com/wakatime/wakatime-cli/pkg/offline.popHeartbeats({0xc000660b80?, 0x40eadf?}, 0x203000?)
	/wakatime-cli/pkg/offline/offline.go:237 +0xb7 fp=0xc0001b3670 sp=0xc0001b3528 pc=0x86b6f7
github.com/wakatime/wakatime-cli/pkg/offline.Sync.func1(0xc00007e840)
	/wakatime-cli/pkg/offline/offline.go:144 +0xca fp=0xc0001b3748 sp=0xc0001b3670 pc=0x86aa6a
github.com/wakatime/wakatime-cli/pkg/offline.WithSync.func1.1({0xc00007e840?, 0xc48320?, 0xc0001b3701?})
	/wakatime-cli/pkg/offline/offline.go:108 +0xd3 fp=0xc0001b37b8 sp=0xc0001b3748 pc=0x86a833
github.com/wakatime/wakatime-cli/pkg/heartbeat.NewHandle.func1({0x0, 0x0, 0x0})
	/wakatime-cli/pkg/heartbeat/heartbeat.go:168 +0xd1 fp=0xc0001b3808 sp=0xc0001b37b8 pc=0x716ad1
github.com/wakatime/wakatime-cli/cmd/offlinesync.SyncOfflineActivity(0x2d1294848?, {0xc000660b80, 0x1d})
	/wakatime-cli/cmd/offlinesync/offlinesync.go:75 +0x3d3 fp=0xc0001b3ad0 sp=0xc0001b3808 pc=0xb9e693
github.com/wakatime/wakatime-cli/cmd/offlinesync.Run(0xe67a20?)
	/wakatime-cli/cmd/offlinesync/offlinesync.go:28 +0x85 fp=0xc0001b3b48 sp=0xc0001b3ad0 pc=0xb9e165
github.com/wakatime/wakatime-cli/cmd.runCmd(0xc00015b380, 0x0, 0x0, 0xd75d60)
	/wakatime-cli/cmd/run.go:268 +0x114 fp=0xc0001b3c10 sp=0xc0001b3b48 pc=0xba1b54
github.com/wakatime/wakatime-cli/cmd.RunCmdWithOfflineSync(0xc0003c58d0?, 0x9e?, 0xb2?, 0x3c?)
	/wakatime-cli/cmd/run.go:244 +0x4b fp=0xc0001b3c48 sp=0xc0001b3c10 pc=0xba19eb
github.com/wakatime/wakatime-cli/cmd.Run(0x0?, 0x0?)
	/wakatime-cli/cmd/run.go:109 +0x5d7 fp=0xc0001b3d70 sp=0xc0001b3c48 pc=0xba0f17
github.com/wakatime/wakatime-cli/cmd.NewRootCMD.func1(0xc000353900?, {0xca59a0?, 0x8?, 0x8?})
	/wakatime-cli/cmd/root.go:29 +0x1d fp=0xc0001b3d90 sp=0xc0001b3d70 pc=0xb9f71d
github.com/spf13/cobra.(*Command).execute(0xc000353900, {0xc0000240a0, 0x8, 0x8})
github.com/spf13/cobra.(*Command).ExecuteC(0xc000353900)
github.com/spf13/cobra.(*Command).Execute(...)
github.com/wakatime/wakatime-cli/cmd.Execute()
	/wakatime-cli/cmd/root.go:260 +0x1e fp=0xc0001b3f70 sp=0xc0001b3f20 pc=0xba08de
main.main()
	/wakatime-cli/main.go:6 +0x17 fp=0xc0001b3f80 sp=0xc0001b3f70 pc=0xba2a97
runtime.main()
	/usr/local/go/src/runtime/proc.go:250 +0x1f8 fp=0xc0001b3fe0 sp=0xc0001b3f80 pc=0x438a38
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0001b3fe8 sp=0xc0001b3fe0 pc=0x466221

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000056fb0 sp=0xc000056f90 pc=0x438df6
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:369
runtime.forcegchelper()
	/usr/local/go/src/runtime/proc.go:302 +0xa5 fp=0xc000056fe0 sp=0xc000056fb0 pc=0x438c85
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000056fe8 sp=0xc000056fe0 pc=0x466221
created by runtime.init.6
	/usr/local/go/src/runtime/proc.go:290 +0x25

goroutine 3 [GC sweep wait]:
runtime.gopark(0x1?, 0x0?, 0x0?, 0x0?, 0x0?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000057790 sp=0xc000057770 pc=0x438df6
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:369
runtime.bgsweep(0x0?)
	/usr/local/go/src/runtime/mgcsweep.go:297 +0xd7 fp=0xc0000577c8 sp=0xc000057790 pc=0x425d97
runtime.gcenable.func1()
	/usr/local/go/src/runtime/mgc.go:178 +0x26 fp=0xc0000577e0 sp=0xc0000577c8 pc=0x41aae6
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0000577e8 sp=0xc0000577e0 pc=0x466221
created by runtime.gcenable
	/usr/local/go/src/runtime/mgc.go:178 +0x6b

goroutine 4 [sleep]:
runtime.gopark(0xc00010a000?, 0x9688d6d3ca7a?, 0x0?, 0x0?, 0xd76428?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000057f30 sp=0xc000057f10 pc=0x438df6
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:369
runtime.(*scavengerState).sleep(0x1334fc0, 0x412f712c00000000)
	/usr/local/go/src/runtime/mgcscavenge.go:468 +0x10f fp=0xc000057fa0 sp=0xc000057f30 pc=0x42400f
runtime.bgscavenge(0x0?)
	/usr/local/go/src/runtime/mgcscavenge.go:626 +0x7f fp=0xc000057fc8 sp=0xc000057fa0 pc=0x4243ff
runtime.gcenable.func2()
	/usr/local/go/src/runtime/mgc.go:179 +0x26 fp=0xc000057fe0 sp=0xc000057fc8 pc=0x41aa86
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000057fe8 sp=0xc000057fe0 pc=0x466221
created by runtime.gcenable
	/usr/local/go/src/runtime/mgc.go:179 +0xaa

goroutine 5 [finalizer wait]:
runtime.gopark(0x0?, 0xd76310?, 0x40?, 0xa0?, 0x2000000020?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000056628 sp=0xc000056608 pc=0x438df6
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:369
runtime.runfinq()
	/usr/local/go/src/runtime/mfinal.go:180 +0x10f fp=0xc0000567e0 sp=0xc000056628 pc=0x419bef
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0000567e8 sp=0xc0000567e0 pc=0x466221
created by runtime.createfing
	/usr/local/go/src/runtime/mfinal.go:157 +0x45

goroutine 19 [GC worker (idle)]:
runtime.gopark(0xc0000527b8?, 0x0?, 0x0?, 0x0?, 0x0?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000052750 sp=0xc000052730 pc=0x438df6
runtime.gcBgMarkWorker()
	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc0000527e0 sp=0xc000052750 pc=0x41cbd1
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0000527e8 sp=0xc0000527e0 pc=0x466221
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/src/runtime/mgc.go:1159 +0x25

goroutine 18 [GC worker (idle)]:
runtime.gopark(0xc000058fb8?, 0x0?, 0x0?, 0x0?, 0xc0000146f0?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000058f50 sp=0xc000058f30 pc=0x438df6
runtime.gcBgMarkWorker()
	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc000058fe0 sp=0xc000058f50 pc=0x41cbd1
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000058fe8 sp=0xc000058fe0 pc=0x466221
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/src/runtime/mgc.go:1159 +0x25

goroutine 9 [GC worker (idle)]:
runtime.gopark(0xc000059fb8?, 0x0?, 0x0?, 0x0?, 0x0?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000059f50 sp=0xc000059f30 pc=0x438df6
runtime.gcBgMarkWorker()
	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc000059fe0 sp=0xc000059f50 pc=0x41cbd1
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000059fe8 sp=0xc000059fe0 pc=0x466221
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/src/runtime/mgc.go:1159 +0x25

goroutine 34 [GC worker (idle)]:
runtime.gopark(0xc0000597b8?, 0x3?, 0xa0?, 0x61?, 0xc000014700?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000059750 sp=0xc000059730 pc=0x438df6
runtime.gcBgMarkWorker()
	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc0000597e0 sp=0xc000059750 pc=0x41cbd1
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0000597e8 sp=0xc0000597e0 pc=0x466221
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/src/runtime/mgc.go:1159 +0x25

goroutine 52 [IO wait]:
runtime.gopark(0x3?, 0xc000182000?, 0x0?, 0x15?, 0xffffffffffffffff?)
	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc0002eb618 sp=0xc0002eb5f8 pc=0x438df6
runtime.netpollblock(0x4b0309?, 0x2eb670?, 0xc0?)
	/usr/local/go/src/runtime/netpoll.go:526 +0xf7 fp=0xc0002eb650 sp=0xc0002eb618 pc=0x432297
internal/poll.runtime_pollWait(0x24e9d2668, 0x72)
	/usr/local/go/src/runtime/netpoll.go:305 +0x89 fp=0xc0002eb670 sp=0xc0002eb650 pc=0x4609c9
internal/poll.(*pollDesc).wait(0xc000378f80?, 0xc000182000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x32 fp=0xc0002eb698 sp=0xc0002eb670 pc=0x4ca0f2
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000378f80, {0xc000182000, 0x1500, 0x1500})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x25a fp=0xc0002eb718 sp=0xc0002eb698 pc=0x4cb45a
net.(*netFD).Read(0xc000378f80, {0xc000182000?, 0xc141e0?, 0xc000670450?})
	/usr/local/go/src/net/fd_posix.go:55 +0x29 fp=0xc0002eb760 sp=0xc0002eb718 pc=0x50f0c9
net.(*conn).Read(0xc000098010, {0xc000182000?, 0x90000c0002eb7e0?, 0x40ee27?})
	/usr/local/go/src/net/net.go:183 +0x45 fp=0xc0002eb7a8 sp=0xc0002eb760 pc=0x51cd45
crypto/tls.(*atLeastReader).Read(0xc00050def0, {0xc000182000?, 0x0?, 0x637719?})
	/usr/local/go/src/crypto/tls/conn.go:787 +0x3d fp=0xc0002eb7f0 sp=0xc0002eb7a8 pc=0x60741d
bytes.(*Buffer).ReadFrom(0xc0000a2278, {0xe67b00, 0xc00050def0})
	/usr/local/go/src/bytes/buffer.go:202 +0x98 fp=0xc0002eb848 sp=0xc0002eb7f0 pc=0x52ea98
crypto/tls.(*Conn).readFromUntil(0xc0000a2000, {0x2134a5d08?, 0xc000098010}, 0xcd574d?)
	/usr/local/go/src/crypto/tls/conn.go:809 +0xe5 fp=0xc0002eb888 sp=0xc0002eb848 pc=0x607605
crypto/tls.(*Conn).readRecordOrCCS(0xc0000a2000, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:616 +0x116 fp=0xc0002ebc10 sp=0xc0002eb888 pc=0x604a56
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:582
crypto/tls.(*Conn).Read(0xc0000a2000, {0xc0002d8000, 0x1000, 0x66d340?})
	/usr/local/go/src/crypto/tls/conn.go:1287 +0x16f fp=0xc0002ebc80 sp=0xc0002ebc10 pc=0x60a6af
bufio.(*Reader).Read(0xc00008ef60, {0xc000166660, 0x9, 0x67b785?})
	/usr/local/go/src/bufio/bufio.go:237 +0x1bb fp=0xc0002ebcb8 sp=0xc0002ebc80 pc=0x632b1b
io.ReadAtLeast({0xe679a0, 0xc00008ef60}, {0xc000166660, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:332 +0x9a fp=0xc0002ebd00 sp=0xc0002ebcb8 pc=0x4a675a
io.ReadFull(...)
	/usr/local/go/src/io/io.go:351
net/http.http2readFrameHeader({0xc000166660?, 0x9?, 0xc000670480?}, {0xe679a0?, 0xc00008ef60?})
	/usr/local/go/src/net/http/h2_bundle.go:1565 +0x6e fp=0xc0002ebd50 sp=0xc0002ebd00 pc=0x6652ee
net/http.(*http2Framer).ReadFrame(0xc000166620)
	/usr/local/go/src/net/http/h2_bundle.go:1829 +0x95 fp=0xc0002ebe00 sp=0xc0002ebd50 pc=0x665b55
net/http.(*http2clientConnReadLoop).run(0xc0002ebf98)
	/usr/local/go/src/net/http/h2_bundle.go:8879 +0x130 fp=0xc0002ebf60 sp=0xc0002ebe00 pc=0x678630
net/http.(*http2ClientConn).readLoop(0xc0000a0180)
	/usr/local/go/src/net/http/h2_bundle.go:8775 +0x6f fp=0xc0002ebfc8 sp=0xc0002ebf60 pc=0x677b4f
net/http.(*http2Transport).newClientConn.func1()
	/usr/local/go/src/net/http/h2_bundle.go:7482 +0x26 fp=0xc0002ebfe0 sp=0xc0002ebfc8 pc=0x670826
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0002ebfe8 sp=0xc0002ebfe0 pc=0x466221
created by net/http.(*http2Transport).newClientConn
	/usr/local/go/src/net/http/h2_bundle.go:7482 +0xaaa

rax    0x4e
rbx    0xc0001b20c0
rcx    0x235f2e597
rdx    0x20000
rdi    0xc5
rsi    0x0
rbp    0x2830fc280
rsp    0x2830fc240
r8     0x1
r9     0x5
r10    0x1
r11    0x206
r12    0x0
r13    0x0
r14    0xc0000061a0
r15    0x2f9c08fea
rip    0x235f2e597
rflags 0x207
cs     0x2b
fs     0x0
gs     0x0

```

**Comment by alanhamlett** [comment]
The error happens on this line:

https://github.com/wakatime/wakatime-cli/blob/27e0a1a0d7135d5aeb721be83c72dd821a5e133b/pkg/offline/offline.go#L237

Maybe a bug in the [bbolt](https://github.com/etcd-io/bbolt) library we use to read that local offline database file?

**Comment by gandarez** [comment]
Are you using the compiled binary distributed [here](https://github.com/wakatime/wakatime-cli/releases/tag/v1.65.2-alpha.2) or compiling by yourself?

**Comment by anoduck** [comment]
@gandarez I have done both, AND both generated the same set of errors. So, it doesn't matter. This might indicate the error is occurring with a library used by the wakatime-cli as @alanhamlett suggested.

**Comment by alanhamlett** [comment]
We just updated that bbolt library, can you see if it's working now using this alpha version: [v1.66.0-alpha.1](https://github.com/wakatime/wakatime-cli/releases/tag/v1.66.0-alpha.1)

**Comment by anoduck** [comment]
@alanhamlett God, Yes! We are back in business. Cooking with peanut oil... or something like that... Booyah!

**Comment by alanhamlett** [comment]
Fixed with #838.
